numpy/add_newdocs.py

```diff
@@ -5395,6 +5395,22 @@ def luf(lamdaexpr, *args, **kwargs):
 
     """)
 
+add_newdoc('numpy.core.multiarray', 'as_index_tuple',
+    """
+    as_index_tuple(index)
+
+    Normalizes an index argument, like that passed to `__getitem__`, into a tuple.
+    Follows the invariant that `x[index]` is identical to `x[as_index_tuple(index)].
```

Comment:
> Stephan, you are a subclasser ;)

To be clear, I only use duck-array types, no subclasses!

I don't know if this would be directly useful (maybe?) -- it's hard to say without a clear description of exactly what it does! In practice I would guess that most folks who write customized indexing code that works mostly but not exactly like NumPy will roll their own normalization (like I did in xarray).